src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Telemetry/VsTelemetryService.cs

```diff
@@ -1,20 +1,142 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
 using System;
+using System.Collections.Generic;
 using System.ComponentModel.Composition;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.ProjectSystem;
 
 namespace Microsoft.VisualStudio.Telemetry
 {
     [Export(typeof(ITelemetryService))]
-    internal class VsTelemetryService : ITelemetryService
+    [Export(typeof(IVsTelemetryService))]
+    internal class VsTelemetryService : IVsTelemetryService, ITelemetryService
     {
+        private const string EventPrefix = "vs/projectsystem/managed/";
+        private const string PropertyPrefix = "VS.ProjectSystem.Managed.";
+        private readonly IUnconfiguredProjectCommonServices _unconfiguredProjectCommonServices;
+        private readonly object _lock =  new object();
+
+        private TelemetryEventCorrelation _telemetryEventCorrelation;
+        private bool _telemetryEventCorrelationInitialized;
+        private Guid _projectGuid;
+
+        [ImportingConstructor]
+        public VsTelemetryService(IUnconfiguredProjectCommonServices unconfiguredProjectCommonServices)
+        {
+            Requires.NotNull(unconfiguredProjectCommonServices, nameof(unconfiguredProjectCommonServices));
+
+            _unconfiguredProjectCommonServices = unconfiguredProjectCommonServices;
+        }
+
+        /// <summary>
+        /// Creates the Project Correlation asset which will be used to correlate all the telemetry events for this project
+        /// identified uniquely through the Project Guid.
+        /// </summary>
+        /// <returns></returns>
+        private async Task<bool> CreateProjectCorrelationAssetAsync()
```

Comment:
I don't understand why we need this data? We already have a project itself - that should be enough to get the rest of the data, including TFMs.